src/core/lib/MagicObject.mjs

```diff
@@ -0,0 +1,134 @@
+/**
+ * An object used by magic to store the input/output criteria for valid operation results. 
+ *
+ * @author n1073645 [n1073645@gmail.com]
+ * @copyright Crown Copyright 2020
+ * @license Apache-2.0
+ */
+import OperationConfig from "../config/OperationConfig.json";
+
+class potentialOps {
+
+    /**
+     * PotentialOps constructor.
+     *
+     * @param prevOp
+     */
+    constructor (prevOp) {
+        if (typeof prevOp === "undefined") {
+            this.inputRegexes = this.generateInputOpPatterns();
+            this.outputRegexes = this.generateOutputOpPatterns();
+        } else {
+            this.inputRegexes = prevOp.getInputRegexes();
+            this.outputRegexes = prevOp.getOutputRegexes();
+        }
+    }
+
+    /**
+     * Sets the output regexes for the operation.
+     *
+     * @param outputRegexes
+     */
+    setOutputRegexes (outputRegexes) {
```

Comment:
Have you taken a look at ES6 getters and setters? Might be a good option here. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/set